sync-team/src/github/api/mod.rs

```diff
@@ -479,3 +479,157 @@ pub(crate) struct RepoSettings {
     pub archived: bool,
     pub auto_merge_enabled: bool,
 }
+
+/// GitHub Repository Ruleset
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub(crate) struct Ruleset {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) id: Option<i64>,
+    pub(crate) name: String,
+    pub(crate) target: RulesetTarget,
+    pub(crate) source_type: RulesetSourceType,
+    pub(crate) enforcement: RulesetEnforcement,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) bypass_actors: Option<Vec<RulesetBypassActor>>,
+    pub(crate) conditions: RulesetConditions,
+    pub(crate) rules: Vec<RulesetRule>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub(crate) enum RulesetTarget {
+    Branch,
+    Tag,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub(crate) enum RulesetSourceType {
+    Repository,
+    Organization,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub(crate) enum RulesetEnforcement {
+    Active,
+    Disabled,
+    Evaluate,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub(crate) struct RulesetBypassActor {
+    pub(crate) actor_id: i64,
+    pub(crate) actor_type: RulesetActorType,
+    pub(crate) bypass_mode: RulesetBypassMode,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub(crate) enum RulesetActorType {
+    Integration,
+    OrganizationAdmin,
+    RepositoryRole,
+    Team,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub(crate) enum RulesetBypassMode {
+    Always,
+    PullRequest,
```

Comment:
<img width="1100" height="1356" alt="Image" src="https://github.com/user-attachments/assets/34433210-ff9b-47b5-adda-3fc72951813c" />

rename_all should be `snake_case`, right?